### PR TITLE
Add fee, cardReference and cardReferenceType to transactions

### PR DESCRIPTION
**Pull request by bramj**

### CODE REVIEW

File: lib/ibanity/api/sandbox/financial_institution_transaction.ex [hunk 1]
```diff
@@ -299,6 +302,9 @@ defmodule Ibanity.Sandbox.FinancialInstitutionTransaction do
       mandate_id: {~w(attributes mandateId), :string},
       creditor_id: {~w(attributes creditorId), :string},
       additional_information: {~w(attributes additionalInformation), :string},
+      fee: {~w(attributes fee), :string},
```
Comment (cmw0513) [inline lib/ibanity/api/sandbox/financial_institution_transaction.ex:305]: Should this be a float?

Comment (thibaultponcelet) [reply lib/ibanity/api/sandbox/financial_institution_transaction.ex:305]: it should use the same type as 'amount' indeed. Decimal may be safer though

Comment (bramj) [reply lib/ibanity/api/sandbox/financial_institution_transaction.ex]: 🤦 auwch that explains the string in Ponto. I've gone and replaced all floats for decimals (as is the case in core), alright?

File: lib/ibanity/api/xs2a/transaction.ex [hunk 2]
```diff
@@ -108,6 +111,9 @@ defmodule Ibanity.Xs2a.Transaction do
       creditor_id: {~w(attributes creditorId), :string},
       digest: {~w(attributes digest), :string},
       additional_information: {~w(attributes additionalInformation), :string},
+      fee: {~w(attributes fee), :string},
```
Comment (cmw0513) [inline lib/ibanity/api/xs2a/transaction.ex:114]: decimal as well?

File: lib/ibanity/api/sandbox/financial_institution_transaction.ex (hunk 1 above)
Comment (bramj) [reply lib/ibanity/api/sandbox/financial_institution_transaction.ex]: Sticking to floats for now, I think I've got them all now

**Pull request merged**